crates/models/src/references.rs

```diff
@@ -239,6 +239,13 @@ impl RelativeUrl {
 #[schemars(example = "CompositeKey::example")]
 pub struct CompositeKey(Vec<JsonPointer>);
 
+impl FromIterator<JsonPointer> for CompositeKey {
```

Comment:
This shouldn't be an implicit conversion IMO.

CompositeKey implies semantics which are not sufficiently conveyed by `some_func(some_iter.into())`.

Reply:
To be clear, I don't think `some_iter.into()` would work. You'd have to call `iter_of_pointers.collect::<CompositeKey>()`. The alternative that uses the existing `new` function would be `CompositeKey::new(iter_of_pointers.collect::<Vec<_>>())`, which I personally don't see as any more readable. Though I'm happy to remove the `FromIterator` impl in favor of the alternative if you prefer.

Reply:
Slightly tweaking my example, `some_func(some_iter.collect())` would work through implicit inference that `some_func` takes a `CompositeKey`, and in my opinion that's too subtle -- there should be an explicit expression in the control flow that "this sequence of pointers is being re-interpreted as a composite key".